File: Data Structures & Algorithms/two-integer-sum/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        for (int i = 0; i < nums.Length; i++) {
            for (int b = i + 1; b < nums.Length; b++) {
                if (nums[i] + nums[b] == target) {
                    return new[] { i, b };
                }
            }
        }
        return null;
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-0.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        lp = 0
        rp = len(matrix) * len(matrix[0]) - 1
        while lp <= rp:
            mid = (rp + lp) // 2
            y = mid//len(matrix[0])
            x = mid - (y * len(matrix[0]))
            val = matrix[y][x]
            if target == val:
                return True
            elif target < val:
                rp = mid-1
            else:
                lp = mid+1
        return False